app/notification/application/map_contents.py

```diff
@@ -157,3 +198,119 @@ def bytes_object_for_questions_answers(
         convert_to_bytes = bytes(stringIO_data, "utf-8")
         bytes_object = BytesIO(convert_to_bytes)
         return bytes_object
+
+    @classmethod
+    def remove_html_tags(cls, answer):
+        """
+        Removes HTML tags from the provided answer and returns the cleaned text.
+
+        Args:
+            answer (str): The answer containing HTML tags.
+
+        Returns:
+            str: The cleaned text with HTML tags removed.
+
+        Example with unordered lis (ul) tags:
+        answer = '<ul><li>Item 1</li><li>Item 2</li></ul>'
+        cleaned_text = remove_html_tags(cls, answer)
+        print(cleaned_text)
+        # Output:
+        #     - Item 1
+        #     - Item 2
+
+        Example with ordered list (ol) tags:
+        answer = '<ol><li>First item</li><li>Second item</li></ol>'
+        cleaned_text = remove_html_tags(cls, answer)
+        print(cleaned_text)
+        # Output:
+        #     1. First item
+        #     2. Second item
+        """
+        try:
+            if answer is None or isinstance(answer, (bool, list)):
+                return answer
+
+            soup = BeautifulSoup(answer, "html.parser")
+            indent = " " * 5
+
+            if soup.find():
+                if soup.ul:
+                    bullet_points = []
+                    for index, li in enumerate(
+                        soup.ul.find_all("li"), start=1
+                    ):
+                        if li.get_text() != "\xa0":
+                            if index > 1:
+                                bullet_points.append(
+                                    f"{indent}- {li.get_text()}"
+                                )
+                            else:
+                                bullet_points.append(f"- {li.get_text()}")
+                    return "\n".join(bullet_points)
+                elif soup.ol:
+                    numbered_list = []
+                    for index, li in enumerate(
+                        soup.ol.find_all("li"), start=1
+                    ):
+                        if li.get_text() != "\xa0":
+                            if index > 1:
+                                numbered_list.append(
+                                    f"{indent}{index}. {li.get_text()}"
+                                )
+                            else:
+                                numbered_list.append(
+                                    f"{index}. {li.get_text()}"
+                                )
+                    return "\n".join(numbered_list)
+                else:
+                    # Handle other HTML tags
+                    cleaned_text = soup.get_text()
+                    cleaned_text = cleaned_text.replace("\xa0", "")
+                    return cleaned_text.strip()
+            else:
+                return answer.strip()
```

Comment:
There is quite a few things I'd suggest to improve this.

First would be using the return early pattern to remove some layers of indentation.

i.e

Instead of:

```python
if soup.find():
    ...
else:
    return answer.strip()
```

We could do:

```python
if not soup.find():
    return answer.strip()
  
...
```

Also the only difference I can see between the chunk of code for `numbered_list` & `bullet_points` is the separator, which is `-` or `.`, all the other code looks the same, could shorten that too:

```python
soup_list = soup.ul or soup.ol
separator = "-" if soup.ul else "."
list_items = []
for index, li in enumerate(soup_list.find_all("li"), start=1):
    if li.get_text() == "\xa0":
        continue

    if index > 1:
        list_items.append(
            f"{indent}{separator} {li.get_text()}"
        )
    else:
        list_items.append(f"{separator} {li.get_text()}")
return "\n".join(list_items)
```

With these suggestions, it would look something like this:

```suggestion
            try:
    
                if answer is None or isinstance(answer, (bool, list)):
                    return answer
    
                soup = BeautifulSoup(answer, "html.parser")
                indent = " " * 5
    
                if not soup.find():
                    return answer.strip()
    
                if not (soup.ul or soup.ol):
                    # Handle other HTML tags
                    cleaned_text = soup.get_text()
                    cleaned_text = cleaned_text.replace("\xa0", "")
                    return cleaned_text.strip()
    
                soup_list = soup.ul or soup.ol
                separator = "-" if soup.ul else "."
                list_items = []
                for index, li in enumerate(soup_list.find_all("li"), start=1):
                    if li.get_text() == "\xa0":
                        continue
    
                    if index > 1:
                        list_items.append(
                            f"{indent}{separator} {li.get_text()}"
                        )
                    else:
                        list_items.append(f"{separator} {li.get_text()}")
                return "\n".join(list_items)
    
            except Exception as e:
                current_app.logger.error(
                    f"Error occurred while processing HTML tag: {answer}", e
                )
                return answer
```

Reply:
Thanks, Thomas, as mentioned on the standup I need to tidy up codes but thank you for highlighting it 😄

Reply:
Ah sorry I missed that!  Let me know if you want a hand, can help write tests too if you'd like

Reply:
Thanks for the tidied code tho I just added an index to the "."
Saved my time 😄